.github/workflows/test.yaml

```diff
@@ -191,28 +193,62 @@ jobs:
       - uses: actions/checkout@v2
         with:
           submodules: true
+      - name: Cache dependencies
+        id: dependency-cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            /tmp/scons_cache
+          key: SCONS_CACHE+${{ hashFiles('.github/workflows/test.yaml') }}
       - name: Build Docker image
-        run: eval "$BUILD"
+        run: eval "$BUILD_LATEST"
       - name: Run unit tests
         run: |
-          $PERSIST "cd /tmp/openpilot && \
-                    scons -j$(nproc) && \
+          $PERSIST_WITH_ONLY_SCONS_CACHE "cd /tmp/openpilot && \
+                    SCONS_CACHE=1 scons -j$(nproc) && \
                     coverage run selfdrive/test/test_fingerprints.py && \
                     $UNIT_TEST common && \
                     $UNIT_TEST opendbc/can && \
                     $UNIT_TEST selfdrive/boardd && \
-                    $UNIT_TEST selfdrive/controls && \
                     $UNIT_TEST selfdrive/monitoring && \
-                    $UNIT_TEST selfdrive/loggerd && \
                     $UNIT_TEST selfdrive/car && \
                     $UNIT_TEST selfdrive/locationd && \
                     $UNIT_TEST selfdrive/athena && \
                     $UNIT_TEST selfdrive/thermald && \
-                    $UNIT_TEST tools/lib/tests"
+                    $UNIT_TEST tools/lib/tests && \
+                    $UNIT_TEST selfdrive/controls"
+                    
       - name: Upload coverage to Codecov
         run: |
           docker commit tmppilot tmppilotci
           $CI_RUN "cd /tmp/openpilot && bash <(curl -s https://codecov.io/bash) -v -F unit_tests"
+                                                                                
+  loggerd_tests:
```

Comment:
the uploader tests was just sleeping for a ridiculous amount of time. fixing in another PR